Distance_3/include/CGAL/squared_distance_3_3.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 1998-2021
+// Utrecht University (The Netherlands),
+// ETH Zurich (Switzerland),
+// INRIA Sophia-Antipolis (France),
+// Max-Planck-Institute Saarbruecken (Germany),
+// and Tel-Aviv University (Israel).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Geert-Jan Giezeman, Andreas Fabri
+
+
+#ifndef CGAL_DISTANCE_3_3_H
+#define CGAL_DISTANCE_3_3_H
+
+#include <CGAL/squared_distance_3_2.h>
+
+#include <CGAL/Point_3.h>
+#include <CGAL/Tetrahedron_3.h>
+
+namespace CGAL {
+
+namespace internal {
+
+template <class K>
+inline
+typename K::FT
+squared_distance(const typename K::Tetrahedron_3 & tet,
+                 const typename K::Point_3 & pt,
+                 const K& k)
+{
+  CGAL_assertion(orientation(tet[0],tet[1],tet[2],tet[3]) == POSITIVE);
+
+  if(orientation(tet[0],tet[1],tet[2],pt) != POSITIVE){
+    const Triangle_3<K> tr = {tet[0], tet[1], tet[2]};
+    return CGAL::squared_distance(tr, pt);
```

Comment:
I don't think this is true. If I take a triangle, and 2 points tet[3] and pt close to each other, far from the triangle, and on opposite sides of the plane containing the triangle, this returns a large distance.

Reply:
you are right. I am ashamed.